src/OSPSuite.R/Services/ConcurrentSimulationRunner.cs

```
@@ -0,0 +1,110 @@
+
+      /// <summary>
+      /// Adds a SimulationBatch to the list of SimulationBatches
+      /// </summary>
+      /// <param name="simulationBatch"></param>
+      void AddSimulationBatch(SimulationBatch simulationBatch);
+
+      void Clear();
+
+      /// <summary>
+      /// Runs all preset settings concurrently
+      /// </summary>
+      /// <returns></returns>
+      SimulationResults[] RunConcurrently();
+   }
+
+   public class ConcurrentSimulationRunner : IConcurrentSimulationRunner
+   {
+      private IConcurrencyManager _concurrentManager;
+
+      public ConcurrentSimulationRunner(IConcurrencyManager concurrentManager)
+      {
+         _concurrentManager = concurrentManager;
+      }
+
+      public SimulationRunOptions SimulationRunOptions { get; set; }
+
+      public int NumberOfCores { get; set; }
+
+      private List<IModelCoreSimulation> _simulations = new List<IModelCoreSimulation>();
+      private List<SimulationBatch> _simulationBatches = new List<SimulationBatch>();
+
+      public void AddSimulation(IModelCoreSimulation simulation)
+      {
+         _simulations.Add(simulation);
+      }
+
+      public void AddSimulationBatch(SimulationBatch simulationBatch)
+      {
+         _simulationBatches.Add(simulationBatch);
+      }
+
+      public void Clear()
+      {
+         _simulations.Clear();
+         _simulationBatches.Clear();
+      }
+
+      public SimulationResults[] RunConcurrently()
+      {
+         //Currently we only allow for running simulations or simulation batches, but not both
+         if (_simulationBatches.Count > 0 && _simulations.Count > 0) 
+            //Temporal Exception. We should allow for mixing both use cases but we need to discuss first
+            throw new Exception("You already have Simulation and SimulationBatch objects and should not mix, please invoke Clear to start adding objects from a fresh start");
+
+         if (_simulations.Count > 0)
```

Simulation.Count = 0
We should probably throw an exception